interface/ceed.c

```diff
@@ -594,8 +594,40 @@ int CeedSetData(Ceed ceed, void *data) {
 /// @addtogroup CeedUser
 /// @{
 
+/**
+  @brief Get the list of avaliable resource names for Ceed contexts
+  Note: The caller is responsible for freeing the resources and priorities arrays,
+          but should not free the contents of the resources array.
+
+  @param[out] n          Number of avaliable resources
+  @param[out] resources  List of avaliable resource names
+  @param[out] priorities Resource name prioritization values, lower is better
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref User
+**/
+// LCOV_EXCL_START
+int CeedRegistryGetList(size_t *n, char ***const resources,
+                        CeedInt **priorities) {
+  int ierr;
+
+  *n = num_backends;
+  ierr = CeedMalloc(*n, resources); CeedChk(ierr);
+  ierr = CeedMalloc(*n, priorities); CeedChk(ierr);
```

Comment:
`priorities` could be optional so you don't have to declare the variable and then free it if you won't use it.